### PR TITLE
December 2020 newsletter draft

### PR DESCRIPTION
**Pull request by milancurcic**
Sorry for the late start on this.

I populated the PR updates for each project. Please review and edit if needed. I will specifically need help from:

* @gklimowicz (Classic and LLVM Flang updates)
* @certik (LFortran updates)
* @LKedward (contributors data update)

As usual, please add your name to the authors list if you add, edit, or review content.

### CODE REVIEW

File: _posts/2020-12-01-Fortran-Newsletter-December-2020.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+* (WIP) [#240](https://github.com/fortran-lang/stdlib/pull/240): Implementation of the `stdlib_stats_distribution` module. It provides probability distribution and statistical functions.
+* (WIP) [#189](https://github.com/fortran-lang/stdlib/pull/189): Initial implementation of sparse matrices.
+
+Don't hesitate to test and review these pull requests!
+
+Otherwise, ongoing discussions continue:
+* [#220](https://github.com/fortran-lang/stdlib/issues/220): API for file system operations: directory manipulation
+* [#241](https://github.com/fortran-lang/stdlib/issues/241): Include a `split` function (202X feature)
+
```
Comment (jvdp1) [inline _posts/2020-12-01-Fortran-Newsletter-December-2020.md:55]:
```suggestion
[#254](https://github.com/fortran-lang/stdlib/issues/254): Proposition to add a logger for debug phases and levels among the different logs.

```

**Review by jvdp1 — COMMENTED** [review]
Thank you @milancurcic. I just suggest  a minor addition for stdlib.

**Comment by milancurcic** [comment]
@certik @gklimowicz do you think you can provide compiler updates by the end of the week? Sorry again for the late notice.

**Comment by gklimowicz** [comment]
Yes, but the probability increases as we approach Friday.

**Comment by milancurcic** [comment]
I'll merge the PR on Monday, 12/07, any additions are welcome until then.

File: _posts/2020-12-01-Fortran-Newsletter-December-2020.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+* Improve the documentation.
+
+The short-term goal of fpm is to make development and installation of Fortran packages with dependencies easier.
+Its long term goal is to build a rich and decentralized ecosystem of Fortran packages and create a healthy
+environment in which new open source Fortran projects are created and published with ease.
+
+## Compilers
+
+### Classic Flang
+
```
Comment (gklimowicz) [inline _posts/2020-12-01-Fortran-Newsletter-December-2020.md:110]: We continue to evaluate and merge pull requests into Classic Flang. Recently merged pull requests into Classic Flang include:
* [PR#883: Flang generated executable does not show result variable of function](https://github.com/flang-compiler/flang/pull/883)
* [PR#933: Updating X-flag entries for internal command line option "-x 49"](https://github.com/flang-compiler/flang/pull/933)
* [PR#939: Publish Arm's internal documentation](https://github.com/flang-compiler/flang/pull/939)
* [PR#941: [DebugInfo] Internal subprogram variable is not accessible for printing in gdb](https://github.com/flang-compiler/flang/pull/941)
* [PR#942: Implement UNROLL(n) directive](https://github.com/flang-compiler/flang/pull/942)
* [PR#943: Enable github Actions for push to master and pull requests to master](https://github.com/flang-compiler/flang/pull/943)
* [PR#945: libpgmath: Stop using pgstdinit.h](https://github.com/flang-compiler/flang/pull/945)
* [PR#946: Call check_member() for PD_is_contiguous](https://github.com/flang-compiler/flang/pull/946)
* [PR#951: Fix for ICE in atomic instruction generation](https://github.com/flang-compiler/flang/pull/951)

Pull requests merged into the supporting projects:
* [classic flang LLVM monorepo PR#5: [Driver] Reduce downstream delta](https://github.com/flang-compiler/classic-flang-llvm-project/pull/5)
* [classic flang LLVM monorepo PR#6: Removing a few CI pipelines](https://github.com/flang-compiler/classic-flang-llvm-project/pull/6)
* [classic flang LLVM monorepo PR#7: Github Actions added to pre-compile artifacts for flang](https://github.com/flang-compiler/classic-flang-llvm-project/pull/7)
* [llvm mirror PR#87: Enable github actions for llvm](https://github.com/flang-compiler/llvm/pull/87)
* [flang-driver PR#94: Enable github actions](https://github.com/flang-compiler/flang-driver/pull/94)

The Classic Flang biweekly call has been set up to discuss issues and plans
for the next pull requests to be validated and merged. Our next calls are Wednesday, November 16 and 30, 8:00 AM Pacific time. The notes from previous calls, upcoming agenda and a link to join the call can be found
[here](https://docs.google.com/document/d/1-OuiKx4d7O6eLEJDBDKSRnSiUO2rgRR-c2Ga4AkrzOI).

File: _posts/2020-12-01-Fortran-Newsletter-December-2020.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+TODO @gklimowicz
+
+### LLVM Flang
+
+TODO @gklimowicz
+
```
Comment (gklimowicz) [inline _posts/2020-12-01-Fortran-Newsletter-December-2020.md:116]: Work continues on LLVM Flang, concentrating on semantics, lowering and runtime. Significant contributions are being made for OpenMP and OpenACC support.

In conjunction with the MLIR-based code from the _fir-dev_ fork (the Fortran IR used for lowering), Flang can compile and run most F77 programs, including the Fortran Compiler Validation Suite (FCVS).

Pat McCormick is (still) working on an RFC for the merge of the lowering code
in the fir-dev fork into LLVM master. (This was interrupted by Supercomputing 2020 and other ECP duties.) The goal is to expedite this in a way that is acceptable to the Flang community, so we can do further work in the single master branch.

Recent updates include:
* Johannes Doerfert has created a web page at https://flang.llvm.org; you can find call and Slack logistics there
* Nichols Romero has an llvm-dev RFC for adding Fortran tests to the llvm-tests project: http://lists.llvm.org/pipermail/llvm-dev/2020-November/146873.html
* Andzrej Warzynski has a flang-dev RFC regarding flang option names: http://lists.llvm.org/pipermail/flang-dev/2020-November/000588.html
* Andzrej Warzynski has a cfe-dev RFC regarding refactoring clang to help flang driver become independent of clang: http://lists.llvm.org/pipermail/cfe-dev/2020-November/067263.html
* Changed representation of CHARACTER data in type system to make more consistent with other types (for arrays)
* Changed COMPLEX expression representation to provide better handling in lowering
* More improvements for supporting Fortran 77 programs
* Implemented runtime support for basic ALLOCATE/DEALLOCATE and further work
* Continued implementation of table-driven runtime for derived types; posted documentation
* Continued implementation of array expression lowering
* Improved error checks on forward references
* More updates to flang driver (option handling; -E can now be used to invoke just the Flang preprocessor)
* OpenACC semantic checks for modifiers on enter/exit data, set directives
* OpenACC lowering (enter/exit data, update, init, shutdown, wait directives)
* OpenMP structure checker updates; semantic checks for copyin clause; schedule class

File: _posts/2020-12-01-Fortran-Newsletter-December-2020.md (hunk 2 above)
Comment (milancurcic) [reply _posts/2020-12-01-Fortran-Newsletter-December-2020.md:110]: Thank you @gklimowicz. The call dates don't seem to be correct--can you check?

File: _posts/2020-12-01-Fortran-Newsletter-December-2020.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+* OpenACC lowering (enter/exit data, update, init, shutdown, wait directives)
+* OpenMP structure checker updates; semantic checks for copyin clause; schedule class
+
+### LFortran
+
+What's new in LFortran:
+
+TODO @certik
+
+You can follow LFortran on Twitter for latest updates: [@lfortranorg](https://twitter.com/lfortranorg).
```
Comment (certik) [inline _posts/2020-12-01-Fortran-Newsletter-December-2020.md:165]: I don't have many updates for LFortran for November, I have been super busy with other things. There will be some good updates for December though.

```suggestion

```

File: _posts/2020-12-01-Fortran-Newsletter-December-2020.md [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+<ul id="page-nav"></ul>
+
+## fortran-lang.org
+
+This month we've had a few updates to the website:
+
+* [#156](https://github.com/fortran-lang/fortran-lang.org/pull/156):
+Updates to the mini-book on building Fortran programs, including the addition of
+short guides on meson and CMake.
```
Comment (certik) [inline _posts/2020-12-01-Fortran-Newsletter-December-2020.md:20]:
```suggestion
short guides on Meson and CMake.
```
?

File: _posts/2020-12-01-Fortran-Newsletter-December-2020.md [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+
+* [#160](https://github.com/fortran-lang/fortran-lang.org/pull/160) (WIP):
+In-depth introduction for Fortran with Make.
+
+[Let us know](https://github.com/fortran-lang/fortran-lang.org/issues) if you have any suggestions for the website and its content.
+We welcome any new contributors to the website and the tutorials page in particular - see the [contributor guide](https://github.com/fortran-lang/fortran-lang.org/blob/master/CONTRIBUTING.md) for how to get started.
+
+## Fortran Standard Library
+
+Here's what's new in stdlib:
```
Comment (certik) [inline _posts/2020-12-01-Fortran-Newsletter-December-2020.md:35]:
```suggestion
Here's what's new in `stdlib`:
```

File: _posts/2020-12-01-Fortran-Newsletter-December-2020.md [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+* [#254](https://github.com/fortran-lang/stdlib/issues/254): Proposition to add a logger for debug phases and levels among the different logs.
+
+The candidate for file system operations to be included in stdlib is being developed by
+[@MarDiehl](https://github.com/MarDiehl) and [@arjenmarkus](https://github.com/arjenmarkus)
+in [this repository](https://github.com/MarDiehl/stdlib_os).
+Please try it out and let us know how it works, if there are any issues, or if the API can be improved.
+
+## Fortran Package Manager
+
+Here's what's new in fpm:
```
Comment (certik) [inline _posts/2020-12-01-Fortran-Newsletter-December-2020.md:64]:
```suggestion
Here's what's new in `fpm`:
```

File: _posts/2020-12-01-Fortran-Newsletter-December-2020.md [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+* (WIP) [#230](https://github.com/fortran-lang/fpm/pull/230),
+        [#261](https://github.com/fortran-lang/fpm/pull/261): Specification of the fpm CLI.
+* (WIP) [#232](https://github.com/fortran-lang/fpm/pull/232): Allowing the `extra` section in fpm.toml.
+* (WIP) [#248](https://github.com/fortran-lang/fpm/pull/248): Refactor backend for incremental rebuilds.
+* (WIP) [#251](https://github.com/fortran-lang/fpm/pull/251): Dependency management.
+* (WIP) [#255](https://github.com/fortran-lang/fpm/pull/255): Setting the compiler and specifying test or app target.
+* (WIP) [#257](https://github.com/fortran-lang/fpm/pull/257): Implement `fpm install`.
+* (WIP) [#260](https://github.com/fortran-lang/fpm/pull/260): Fix CI to test release build.
+
+fpm is still in early development and we need as much help as we can get.
```
Comment (certik) [inline _posts/2020-12-01-Fortran-Newsletter-December-2020.md:95]:
```suggestion
`fpm` is still in early development and we need as much help as we can get.
```

**Review by certik — APPROVED** [review]
I left some minor comments. It looks good!

File: _posts/2020-12-01-Fortran-Newsletter-December-2020.md (hunk 2 above)
Comment (gklimowicz) [reply _posts/2020-12-01-Fortran-Newsletter-December-2020.md:110]: Done. The dates were right, but the month was wrong. ;-)

**Comment by milancurcic** [comment]
#build_preview

**Comment by LKedward** [comment]
Apologies, it looks like the build preview action is [broken](https://github.com/fortran-lang/fortran-lang.org/runs/1511567789?check_suite_focus=true) due to a previous deprecation in github actions.

**Comment by milancurcic** [comment]
Thank you all! I will go ahead and merge.

The build preview CI failed because `set-env` is not allowed anymore. I will open a separate issue for that, but for now I will just go ahead with this as the newsletter seems fine from looking at the md file.

**Pull request merged**